  <launch>
  
  <arg name="file_name" default="file"/>
  <arg name="playing_back" default="false"/>
  <arg name="record" default="false" />

  <arg name="uav_odometry" value="uav_odometry"/>
  <arg name="reference_position" value="reference_position"/>
  <arg name="fa_disturbance_estimate" value="fa_disturbance_estimate"/>

   <!-- SYSTEM -->
   <group unless="$(arg playing_back)">
     <include file="$(find quad_control)/launch/firefly_example.launch">
       <!-- all vars that included.launch requires must be set -->
     </include>
   </group>
 
   <group ns="firefly1">
   <!-- <group ns="firefly"> -->
 
     <!-- remap topic name -->
     <remap from="$(arg uav_odometry)" to="/firefly/ground_truth/odometry"/>
     <!-- <remap from="$(arg uav_odometry)" to="ground_truth/odometry"/> -->
     <remap from="firefly_motor_speed" to="/firefly/command/motor_speed"/>
     <!-- <remap from="firefly_motor_speed" to="command/motor_speed"/> -->
 
    <param name="uav_odometry" type="str" value="$(arg uav_odometry)"/>
    <param name="reference_position" type="str" value="$(arg reference_position)"/>
    <param name="fa_disturbance_estimate" type="str" value="$(arg fa_disturbance_estimate)"/>
    <param name="playing_back" type="bool" value="$(arg playing_back)"/>

    <!-- Controller Node -->
    <include file="$(find quad_control)/launch/firefly_parameters.launch" />
    <include file="$(find quad_control)/launch/firefly_control_parameters.launch" />
    <node pkg="quad_control" name="controller_firefly" type="cycle_quad_control_mission.py" output="screen"/>

    <group if="$(arg playing_back)">
      <node pkg="rosbag" type="play" name="rosbag_record"
        args="$(find quad_control)/experimental_data/rosbag/$(arg file_name).bag"
        if="$(arg record)"/>
    </group>

    <group unless="$(arg playing_back)">
      <node pkg="rosbag" type="record" name="rosbag_record"
        args="-O $(find quad_control)/experimental_data/rosbag/$(arg file_name) ServiceSequencer $(arg uav_odometry) uav_roll_pitch_yawrate_thrust $(arg fa_disturbance_estimate) $(arg reference_position)"
        if="$(arg record)"/>  
    </group>

    <include file="$(find quad_control)/launch/gui.launch">
      <!-- all vars that included.launch requires must be set -->
      <arg name="gui_off" value="$(arg playing_back)"/>
    </include>

  </group>

</launch>